extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeChannelLinkHandlerFactory.java

```diff
@@ -77,35 +92,38 @@ public String getId(final String url) throws ParsingException {
             if (!Utils.isHTTP(urlObj) || !(YoutubeParsingHelper.isYoutubeURL(urlObj)
                     || YoutubeParsingHelper.isInvidioURL(urlObj)
                     || YoutubeParsingHelper.isHooktubeURL(urlObj))) {
-                throw new ParsingException("the URL given is not a Youtube-URL");
+                throw new ParsingException("The URL given is not a YouTube URL");
             }
 
-            // remove leading "/"
+            // Remove leading "/"
             path = path.substring(1);
+
             String[] splitPath = path.split("/");
 
-            // Handle custom short channel URLs like youtube.com/yourcustomname
-            if (isCustomShortChannelUrl(splitPath)) {
+            if (isHandle(splitPath)) {
+                // Handle YouTube handle URLs like youtube.com/@yourhandle
+                return splitPath[0];
+            } else if (isCustomShortChannelUrl(splitPath)) {
+                // Handle custom short channel URLs like youtube.com/yourcustomname
                 path = "c/" + path;
                 splitPath = path.split("/");
             }
 
-            if (!path.startsWith("user/")
-                    && !path.startsWith("channel/")
+            if (!path.startsWith("user/") && !path.startsWith("channel/")
                     && !path.startsWith("c/")) {
-                throw new ParsingException("the URL given is neither a channel nor an user");
+                throw new ParsingException(
+                        "The given URL is not a channel, a user or a handle URL");
             }
 
             final String id = splitPath[1];
 
-            if (id == null || !id.matches("[A-Za-z0-9_-]+")) {
-                throw new ParsingException("The given id is not a Youtube-Video-ID");
+            if (isBlank(id)) {
```

Comment:
Do you think we should use something else to match all custom names of channels or not?

Reply:
Is it possible to have a channel ID that does not match the old pattern, e.g. by consisting of non latin characters?

Reply:
Ok. dumb question. you put one into your tests... https://www.youtube.com/c/%EB%85%B8%EB%A7%88%EB%93%9C%EC%BD%94%EB%8D%94NomadCoders